executor/src/witgen/query_processor.rs

```diff
@@ -103,32 +103,55 @@ impl<'a, T: FieldElement> SymbolLookup<'a, T> for Symbols<'a, T> {
         name: &'a str,
         type_args: Option<Vec<Type>>,
     ) -> Result<Arc<Value<'a, T>>, EvalError> {
-        Definitions::lookup_with_symbols(
-            &self.fixed_data.analyzed.definitions,
-            name,
-            type_args,
-            self,
-        )
+        match self.fixed_data.analyzed.intermediate_columns.get(name) {
+            // Intermediate polynomials (which includes challenged) are not inlined in hints,
+            // so we need to look them up here.
+            Some((symbol, expressions)) => {
```

Comment:
Can you assert that type_args is None here?

Reply:
Hmm, then the assertion fails, for the challenge I get `Some([])`! That should be fine though, right? I added an assert that it is none or empty.